### PR TITLE
EVG-3660: convert user and usermanager implementations

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: auth/auth_test.go [hunk 1]
```diff
@@ -32,7 +33,7 @@ func TestLoadUserManager(t *testing.T) {
 				Naive:  nil,
 				Github: &g}
 			_, err := LoadUserManager(a)
-			So(err, ShouldBeNil)
+			So(err, ShouldNotBeNil)
```
Comment (bsamek) [inline auth/auth_test.go]: Seems like this should be nil.

File: model/user_lifecycle.go [hunk 2]
```diff
@@ -20,6 +20,14 @@ func SetUserAPIKey(userId, newKey string) error {
 	return user.UpdateOne(bson.M{user.IdKey: userId}, update)
 }
 
+func FindUserByID(id string) (*user.DBUser, error) {
+	t, err := user.FindOne(user.ById(id))
+	if err != nil {
+		return nil, err
```
Comment (bsamek) [inline model/user_lifecycle.go]: wrap?

File: auth/user.go [hunk 3]
```diff
@@ -1,60 +1,16 @@
 package auth
 
-import (
-	"context"
-	"net/http"
-)
-
-// User describes an Evergreen user and is returned by a UserManager.
```
Comment (bsamek) [inline auth/user.go]: Why remove these interfaces?

Comment (tychoish) [reply auth/user.go]: Because they're replaced by gimlet ones :)

File: auth/auth_test.go (hunk 1 above)
Comment (tychoish) [reply auth/auth_test.go]: this is a really silly test of a factory function, and the function looks right and this is the obvious test. I've fixed some of the wording.

File: model/user_lifecycle.go (hunk 2 above)
Comment (tychoish) [reply model/user_lifecycle.go]: done

**Pull request merged**